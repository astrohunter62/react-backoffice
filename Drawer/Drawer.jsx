import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from 'material-ui/styles'
import MaterialDrawer from 'material-ui/Drawer'
import Divider from 'material-ui/Divider'
import IconButton from 'material-ui/IconButton'
import ChevronLeftIcon from 'material-ui-icons/ChevronLeft'

import Menu from '../Menu'

const drawerWidth = 280

const styles = (theme) => ({
  drawerPaper: {
    position: 'relative',
    height: '100%',
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    height: theme.spacing.unit * 8
  },
})

const Drawer = ({
  data,
  open,
  handleDrawerClose,
  redirectTo,
  classes
}) => {
  return (
    <MaterialDrawer
      type="persistent"
      classes={{
        paper: classes.drawerPaper,
      }}
      anchor="left"
      open={open}
    >
      <div className={classes.drawerInner}>
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>

        <Divider />

        <Menu data={data} redirectTo={redirectTo} />
      </div>
    </MaterialDrawer>
  )
}

Drawer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  open: PropTypes.bool.isRequired,
  handleDrawerClose: PropTypes.func.isRequired,
  redirectTo: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Drawer)
