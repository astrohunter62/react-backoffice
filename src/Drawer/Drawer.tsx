import React from "react";
import {
  Drawer as MaterialDrawer,
  Divider,
  IconButton,
  withStyles,
} from "@material-ui/core";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import Menu from "../Menu";
import { MenuDataItem } from "../Menu/Menu";
const drawerWidth = 280;
const styles = (theme: any) => ({
  drawerPaper: {
    position: "relative",
    height: "100%",
    width: drawerWidth,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    height: theme.spacing(8),
  },
});

type DrawerProps = {
  data: MenuDataItem[];
  isOpen?: boolean;
  onClose: (...args: any[]) => any;
  redirectTo: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};

const Drawer: React.SFC<DrawerProps> = ({
  data,
  isOpen,
  onClose,
  redirectTo,
  classes,
}) => (
  <MaterialDrawer
    variant="persistent"
    classes={{
      paper: classes.drawerPaper,
    }}
    anchor="left"
    open={isOpen}
  >
    <div className={classes.drawerInner}>
      <div className={classes.drawerHeader}>
        <IconButton onClick={onClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>

      <Divider />

      <Menu data={data} redirectTo={redirectTo} />
    </div>
  </MaterialDrawer>
);
Drawer.defaultProps = {
  isOpen: false,
};
export default withStyles(styles as any)(Drawer);
