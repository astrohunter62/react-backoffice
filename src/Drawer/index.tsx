import React from "react";
import {
  Drawer as MaterialDrawer,
  Divider,
  IconButton,
  Theme,
} from "@material-ui/core";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import Menu from "../Menu";
import { MenuDataItem } from "../Menu";
import { makeStyles } from "@material-ui/styles";

export const DRAWER_WIDTH = 240;
const useStyles = makeStyles((theme: Theme) => ({
  drawerPaper: {
    width: DRAWER_WIDTH,
  },

  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    height: theme.spacing(8),
  },
}));

type DrawerProps = {
  data: MenuDataItem[];
  isOpen?: boolean;
  onClose: (...args: any[]) => any;
  redirectTo: (...args: any[]) => any;
};

const Drawer: React.SFC<DrawerProps> = ({
  data,
  isOpen = false,
  onClose,
  redirectTo,
}) => {
  const classes = useStyles();

  return (
    <MaterialDrawer
      variant="persistent"
      classes={{
        paper: classes.drawerPaper,
      }}
      anchor="left"
      open={isOpen}
    >
      <div className={classes.drawerHeader}>
        <IconButton onClick={onClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>

      <Divider />

      <Menu data={data} redirectTo={redirectTo} />
    </MaterialDrawer>
  );
};

export default Drawer;
