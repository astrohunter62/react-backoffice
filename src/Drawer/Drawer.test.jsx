import React from 'react'
import renderer from 'react-test-renderer'

import menuData from '../../__tests__/data/menu'

import Drawer from './'

it('renders correctly', () => {
  const tree = renderer
    .create(<Drawer
      isOpen
      data={menuData}
      onClose={() => {}}
      redirectTo={() => {}}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
