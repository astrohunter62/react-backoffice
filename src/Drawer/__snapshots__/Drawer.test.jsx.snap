// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiDrawer-docked-3"
>
  <div
    className="MuiPaper-root-14 MuiPaper-shadow0-16 MuiDrawer-paper-4 Drawer-drawerPaper-1 MuiDrawer-paperAnchorLeft-5 MuiDrawer-paperAnchorDockedLeft-9"
    direction="right"
    style={Object {}}
  >
    <div
      className={undefined}
    >
      <div
        className="Drawer-drawerHeader-2"
      >
        <button
          className="MuiButtonBase-root-47 MuiIconButton-root-41"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-46"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-49 MuiSvgIcon-fontSize-55"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-56"
          />
        </button>
      </div>
      <hr
        className="MuiDivider-root-63 MuiDivider-default-65"
      />
      <div
        className="Menu-root-68"
      >
        <ul
          className="MuiList-root-69 MuiList-padding-70"
        />
      </div>
    </div>
  </div>
</div>
`;
